### PR TITLE
style: simplify string formatting for readability

### LINKED ISSUE #4000: Some statements can be simplified for readability
We can simplify some statements like so to improve code readability and maintainability.

```rust
            X11Error::Xlib(e) => write!(f, "Xlib error: {e}"),
            X11Error::Connect(e) => write!(f, "X11 connection error: {e}"),
            X11Error::Connection(e) => write!(f, "X11 connection error: {e}"),
            X11Error::XidsExhausted(e) => write!(f, "XID range exhausted: {e}"),
            X11Error::GetProperty(e) => write!(f, "Failed to get X property {e}"),
            X11Error::X11(e) => write!(f, "X11 error: {e:?}"),
            X11Error::UnexpectedNull(s) => write!(f, "Xlib function returned null: {s}"),
```

### PR DESCRIPTION
**Pull request by hamirmahal**
Closes #4000

- [ ] Tested on all platforms changed
- [x] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Review by hamirmahal — COMMENTED** [review]
This shouldn't cause any user-facing changes.

The goal here is to improve code readability and developer productivity.

**Comment by madsmtm** [comment]
It sounds like you're using an automated tool to find these? From a quick look I found at least the similar https://github.com/tembo-io/pgmq/pull/341, https://github.com/stalwartlabs/mail-server/pull/909, https://github.com/mullvad/pfctl-rs/pull/119 and https://github.com/amodm/webbrowser-rs/pull/91.

It would be really nice if more context was given (that these were found in an automated fashion, if so), otherwise it feels disingenuous.

**Comment by hamirmahal** [comment]
Oh, I found these with `clippy`.

**Comment by madsmtm** [comment]
Ah, makes sense. Would you be up for adding the specific lints to `lib.rs`, then?

**Comment by hamirmahal** [comment]
Sure! I'll take a look at that right now.

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![warn(clippy::uninlined_format_args)]
```
Comment (madsmtm) [inline src/lib.rs:1]: Nit: Lints are down below the doc comment

**Review by madsmtm — APPROVED** [review]
Thanks

**Comment by hamirmahal** [comment]
You're welcome.

**Pull request merged**